sdk/generic-sdk-core/generic-core-http-okhttp/src/main/java/com/generic/core/http/okhttp/OkHttpHttpClient.java

```diff
@@ -138,28 +140,17 @@ private static long getRequestContentLength(BinaryData content, Headers headers)
         return contentLength;
     }
 
-    private static Response<?> toResponse(HttpRequest request, okhttp3.Response response, boolean eagerlyReadResponse,
-                                          boolean ignoreResponseBody, boolean eagerlyConvertHeaders) throws IOException {
-        /*// For now, eagerlyReadResponse and ignoreResponseBody works the same.
-        if (ignoreResponseBody) {
-            ResponseBody body = response.body();
-
-            if (body != null) {
-                if (body.contentLength() > 0) {
-                    LOGGER.log(LogLevel.WARNING, () -> "Received HTTP response body when one wasn't expected. "
-                        + "Response body will be ignored as directed.");
-                }
-
-                body.close();
-            }
-
+    private static Response<?> toResponse(HttpRequest request, okhttp3.Response response,
+                                          ResponseBodyHandling responseBodyHandling, boolean eagerlyConvertHeaders) throws IOException {
+        if (responseBodyHandling == IGNORE) {
             return new OkHttpResponse(response, request, eagerlyConvertHeaders, EMPTY_BODY);
-        }*/
-
-        // Use a buffered response when we are eagerly reading the response from the network and the body isn't empty.
-        if (eagerlyReadResponse || ignoreResponseBody) {
+        } else if (responseBodyHandling == DESERIALIZE) {
             try (ResponseBody body = response.body()) {
-                byte[] bytes = (body != null) ? body.bytes() : EMPTY_BODY;
+                byte[] bytes = body == null
+                    ? null
+                    : body.contentLength() == 0
+                        ? null
+                        : body.bytes();
```

Comment:
Was there a reason to change the previous behavior? Also, for readability, let's not have a ternary in a ternary, it's very easy to mess up operation orders and break code in these cases.

Reply:
Using `null` with these new changes is used to tell `OkHttpResponse` it needs to read the `ResponseBody` contents.